frontend/packages/erd-core/src/stores/userEditing/store.ts

```diff
@@ -52,3 +52,18 @@ subscribe(userEditingStore.hiddenNodeIds, async () => {
 
   window.history.pushState({}, '', url)
 })
+
+let previousShowMode = userEditingStore.showMode
```

Comment:
Why should the previous value be retained?
I thought that since it is executed when there is a change in the value, it would never match the previous value.

Reply:
In this case, since we're using the following code, we'll also subscribe to changes other than userEditingStore.showMode.💦
```
subscribe(userEditingStore, () => {
```

not
```
subscribe(userEditingStore.showcase, () => {
```
↑ I tried this approach as well, but I couldn't resolve the following error.
```
Argument of type 'string' is not assignable to parameter of type 'object'.
```

Therefore, by using ``previousShowMode``, we update the URL query parameter **only when userEditingStore.showMode changes**. This prevents unnecessary calls to ``window.history.pushState.``

Reply:
I see, for that purpose, I thought it would be a good to try valtio's subscribeKey first!
https://valtio.dev/docs/api/utils/subscribeKey

If this still does not work, the current code looks good!

Reply:
Thank you! I'll give it a try🙏.

Reply:
I was able to fix the code using the subscribeKey! Thank you.
https://github.com/liam-hq/liam/pull/452/commits/fc6a5ddc1f46be4b870c8b9702b132e6381dabae